src/components/noticias/NoticiaForm.jsx

```diff
@@ -252,40 +254,55 @@ class NoticiaForm extends Component {
     newState.vimeo['id'] = id;
 
     if(id !== undefined && id !== null && id !== '') { // If id has a value
-
       // Call to get the video data
-      const videoData = getVimeoData(id, response => {
+      getVimeoData(id, this.handleVimeoData);
 
-        if (response.error) { // Something wetn wrong :(
-          newState.vimeo['sources'] = undefined;
-          toastr.warning('Error', 'El ID de Vimeo no es correcto', ToastrOptionsError);
-        }
+    } else {
+      newState.vimeo = {
+        id: ''
+      };
+    }
 
-        if(response.body) { // We got the video data
+    this.setState(newState);
 
-          toastr.success('Éxito', `Agregado el video "${response.body.name}"`, ToastrOptionsSuccess);
+  }
 
-          // Build an object of sources
-          response.body.files.forEach( source => {
-            const key = source.width !== undefined ? source.width : 'original';
-            const thumb = response.body.pictures.sizes.find( size => size.width === source.width);
+  handleVimeoData(response) {
+    let newState = this.state;
 
-            if(thumb) {
-              source.thumb = thumb.link_with_play_button;
-            }
+    if (response.error) { // Something wetn wrong :(
+      newState.vimeo['sources'] = {};
 
-            newState.vimeo['sources'][key] = source;
-          });
+      const message = JSON.parse(response.error.message);
 
-        }
+      if(message.error === 'The requested video could not be found') {
```

Comment:
this could check against code rather than string (but not sure if that is better or worse)

Reply:
here is a possible error that we should translate and handle:

```
Error: {
    "error": "Unable to locate video. Please get in touch with the app's creator.",
    "link": null,
    "developer_message": "The video is not owned by the user.",
    "error_code": 5001
}
```

Reply:
Actually if a non existent `id` is used, only `error` is available. A generic warning should be fine then :/